src/Command/DrupalUpdate/DrupalPackagesInfo.php

```diff
@@ -0,0 +1,167 @@
+<?php
+
+namespace Acquia\Cli\Command\DrupalUpdate;
+
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Finder\Finder;
+
+class DrupalPackagesInfo{
+  /**
+   * @var array
+   */
+  private array $packageData=[];
+
+  /**
+   * @var bool
+   */
+  private bool $isCoreUpdated;
+
+  /**
+   * @var DrupalOrgClient
+   */
+  private DrupalOrgClient $drupalOrgClient;
+
+  /**
+   * @return DrupalOrgClient
+   */
+  public function getDrupalOrgClient(): DrupalOrgClient {
+    return $this->drupalOrgClient;
+  }
+
+  /**
+   * @param DrupalOrgClient $drupalOrgClient
+   */
+  public function setDrupalOrgClient(DrupalOrgClient $drupalOrgClient): void {
+    $this->drupalOrgClient = $drupalOrgClient;
+  }
+
+  /**
+   * @param bool $isCoreUpdated
+   */
+  public function setIsCoreUpdated(bool $isCoreUpdated): void {
+    $this->isCoreUpdated = $isCoreUpdated;
+  }
+
+  /**
+   * @return array
+   */
+  public function getPackageData(): array {
+    return $this->packageData;
+  }
+
+  public function __construct(InputInterface $input, OutputInterface $output) {
+    $this->setIsCoreUpdated(FALSE);
+    $this->setDrupalOrgClient(new DrupalOrgClient($input, $output));
+  }
+
+  /**
+   * @return mixed
+   */
+  protected function determineCorePackageVersion() {
+    $drupal_boostrap_inc_path = getcwd() . '/docroot/includes/bootstrap.inc';
+    if (file_exists($drupal_boostrap_inc_path)) {
+      $boostrap_inc_file_contents = file_get_contents($drupal_boostrap_inc_path);
+      preg_match("/define\(\s*'([^']*)'\s*,\s*'([^']*)'\s*\)/i", $boostrap_inc_file_contents, $constraint_matches);
+      if ((count($constraint_matches) > 2) && ($constraint_matches[1] == 'VERSION')) {
+        return $constraint_matches[2];
+      }
+    }
+    return '';
+  }
+
+  /**
+   * @return array
+   */
+  public function getInfoFilesList() {
+    $finder = new Finder();
+    $finder->files()->in(getcwd())->name('*.info');
+    $info_package_files = [];
+    foreach ($finder as $file) {
+      $package_dir_path = $file->getRealPath();
+      $package_dir = basename($package_dir_path);
+      if (isset($info_package_files[$package_dir])) {
+        $directory_temp_path = $info_package_files[$package_dir] . "," . $package_dir_path;
+        $info_package_files[$package_dir] = $directory_temp_path;
+      }
+      else {
+        $info_package_files[$package_dir] = $package_dir_path;
+      }
+    }
+    return $info_package_files;
+  }
+
+  /**
+   * @param $info_packages_file_path
+   * @throws AcquiaCliException
+   */
+  public function getPackageDetailInfo($info_packages_file_path) {
+    foreach ($info_packages_file_path as $package_name => $package_path) {
+      if ( strpos($package_path, "," ) !== FALSE ) {
```

Comment:
Can be replaced by `str_contains()` - Also extra whitespaces between parenthesis are not part of the typical coding style of this project.